### PR TITLE
chore: replace chartjs-adapter-moment

### LINKED ISSUE #5042: Replace chartjs-adapter-moment with solution including days
## Abstract

Make sure that moment.js is not pulled as a transient dependency

## Motivation

At some point last year we replaced moment.js with day.js https://github.com/rotki/rotki/pull/3149 the reasoning can be found in the PR.

I failed to notice initially but we use the `chartjs-adapter-moment` which pulls moment.js as a dependency. This means that at the moment we ship both day.js and moment.js. 

The adapter itself is pretty trivial:

https://github.com/chartjs/chartjs-adapter-moment/blob/cd13f90dc1afb73dfd4cee00b6f2ade7bd874d44/src/index.js#L6

Here is one that is using dayjs instead.

https://gitlab.com/mmillerbkg/chartjs-adapter-dayjs/-/blob/master/src/index.ts

## Specification

Make the adapter internally in our codebase and remove the external dependency.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #5042

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/package.json [hunk 1]
```diff
@@ -64,7 +63,6 @@
     "@mdi/font": "7.0.96",
     "@namics/stylelint-bem": "6.3.4",
     "@pinia/testing": "0.0.14",
-    "@types/chart.js": "2.9.37",
```
Comment (kelsos) [inline frontend/app/package.json:67]: So chart.js 3 proves it's own types?

File: frontend/app/src/utils/chartjs-adapter-dayjs.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+dayjs.extend(AdvancedFormat);
+
+dayjs.extend(QuarterOfYear);
+
+dayjs.extend(LocalizedFormat);
+
+dayjs.extend(CustomParseFormat);
+
+dayjs.extend(isoWeek);
```
Comment (kelsos) [inline frontend/app/src/utils/chartjs-adapter-dayjs.ts:27]: nitpick: we don't really need an empty space after every extend.

File: frontend/app/src/utils/chartjs-adapter-dayjs.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  hour: 'hA',
+  day: 'MMM D',
+  week: 'll',
+  month: 'MMM YYYY',
+  quarter: '[Q]Q - YYYY',
+  year: 'YYYY'
+};
+
+_adapters._date.override({
+  //_id: 'dayjs', //DEBUG,
```
Comment (kelsos) [inline frontend/app/src/utils/chartjs-adapter-dayjs.ts:43]: Can we either remove this comment, or if needed uncomment it.

**Review by kelsos — APPROVED** [review]
I have some minor comments but after fixing them we can merge.

File: frontend/app/package.json (hunk 1 above)
Comment (lukicenturi) [reply frontend/app/package.json:67]: Yeah it is

**Pull request merged**